tests/components/netgear_lte/test_sensor.py

```diff
@@ -1,56 +1,38 @@
-    PERCENTAGE,
-    SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
-    UnitOfInformation,
-)
+from syrupy.assertion import SnapshotAssertion
+
 from homeassistant.core import HomeAssistant
 
 
-@pytest.mark.usefixtures("setup_integration", "entity_registry_enabled_by_default")
-async def test_sensors(hass: HomeAssistant) -> None:
+async def test_sensors(
+    hass: HomeAssistant,
+    entity_registry_enabled_by_default: None,
+    setup_integration: None,
+    snapshot: SnapshotAssertion,
+) -> None:
     """Test for successfully setting up the Netgear LTE sensor platform."""
-    state = hass.states.get("sensor.netgear_lm1200_cell_id")
-    assert state.state == "12345678"
-    state = hass.states.get("sensor.netgear_lm1200_connection_text")
-    assert state.state == "4G"
-    state = hass.states.get("sensor.netgear_lm1200_connection_type")
-    assert state.state == "IPv4AndIPv6"
-    state = hass.states.get("sensor.netgear_lm1200_current_band")
-    assert state.state == "LTE B4"
-    state = hass.states.get("sensor.netgear_lm1200_service_type")
-    assert state.state == "LTE"
-    state = hass.states.get("sensor.netgear_lm1200_radio_quality")
-    assert state.state == "52"
-    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == PERCENTAGE
-    state = hass.states.get("sensor.netgear_lm1200_register_network_display")
-    assert state.state == "T-Mobile"
-    state = hass.states.get("sensor.netgear_lm1200_rx_level")
-    assert state.state == "-113"
-    assert (
-        state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
-        == SIGNAL_STRENGTH_DECIBELS_MILLIWATT
-    )
-    state = hass.states.get("sensor.netgear_lm1200_sms")
-    assert state.state == "1"
-    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == "unread"
-    state = hass.states.get("sensor.netgear_lm1200_sms_total")
-    assert state.state == "1"
-    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == "messages"
-    state = hass.states.get("sensor.netgear_lm1200_tx_level")
-    assert state.state == "4"
-    assert (
-        state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
-        == SIGNAL_STRENGTH_DECIBELS_MILLIWATT
-    )
-    state = hass.states.get("sensor.netgear_lm1200_upstream")
-    assert state.state == "LTE"
-    state = hass.states.get("sensor.netgear_lm1200_usage")
-    assert state.state == "40.5"
-    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == UnitOfInformation.MEBIBYTES
-    assert state.attributes[ATTR_DEVICE_CLASS] == SensorDeviceClass.DATA_SIZE
+    assert hass.states.get("sensor.netgear_lm1200_cell_id") == snapshot
+
+    assert hass.states.get("sensor.netgear_lm1200_connection_text") == snapshot
+
+    assert hass.states.get("sensor.netgear_lm1200_connection_type") == snapshot
+
+    assert hass.states.get("sensor.netgear_lm1200_current_band") == snapshot
+
+    assert hass.states.get("sensor.netgear_lm1200_service_type") == snapshot
+
+    assert hass.states.get("sensor.netgear_lm1200_radio_quality") == snapshot
```

Comment:
You can also read them from the entity registry, I have such setup in Withings, maybe check that out :)